src/sign-typed-data.ts

```diff
@@ -262,18 +278,50 @@ function encodeField(
 
   if (type === 'address') {
     if (typeof value === 'number') {
-      return ['address', padStart(numberToBytes(value), 20)];
+      // Unnecessary casting - Request value could have been a hex string
+      return [
+        'address',
+        padStart(numberToBytes(value), 20),
+        {
+          typeOfFeedback: MetricFeedbackType.VIOLATION,
+          type,
+          name,
+          actual: 'number',
+        },
+      ];
     } else if (isStrictHexString(value)) {
-      return ['address', add0x(value)];
+      // Unnecessary casting - Request value could have been a hex string - This is already checking for a hex string, so add0x is redundant
+      return [
+        'address',
+        add0x(value),
+        {
+          typeOfFeedback: MetricFeedbackType.UNNECESSARY_CASTING,
+          type,
+          name,
+          actual: 'hex string',
+        },
+      ];
     } else if (typeof value === 'string') {
+      // Necessary casting - See note above function - TLDR: It's here for backwards compatibility
       return ['address', reallyStrangeAddressToBytes(value).subarray(0, 20)];
     }
   }
 
   if (type === 'bool') {
-    return ['bool', Boolean(value)];
+    // Unnecessary casting - Request value could have been a boolean
```

Comment:
> Unnecessary casting - Request value could have been a boolean

maybe I am not understanding the comment fully. It seems we expect it to be boolean but it could be another type.  maybe we can mention is it unnecessary because we should expect explicit boolean types: `true` and `false` only. other types, even `"true"` or `"false"` should be rejected.

I guess if we did make this stricter, we could consider adding support by explicitly casting "true"/"false" or 1/0.